doc/src/content/xdocs/spec.xml

```diff
@@ -1155,8 +1168,9 @@
       <title>Parsing Canonical Form for Schemas</title>
 
       <p>One of the defining characteristics of Avro is that a reader
-      is assumed to have the "same" schema used by the writer of the
-      data the reader is reading.  This assumption leads to a data
+      must read serialized data with the "same" schema used by the
```

Comment:
I think this could simply say that the reader "must use the 'same' schema used by the writer when reading".